### PR TITLE
Add support for optional schema role generation

### PR DESCRIPTION
**Pull request by acaruso7**
## Overview

By default, SnowDDL autogenerates 3 schema roles for every schema defined in the config:
* `<db>__<schema>__OWNER__S_ROLE`
* `<db>__<schema>__READ__S_ROLE`
* `<db>__<schema>__WRITE__S_ROLE`

Reference: https://docs.snowddl.com/guides/role-hierarchy#schema-roles

These roles are granted a number of pre-defined, low-level privileges

In some cases, these roles may not always be desired. This PR extends the `SchemaRoleResolver` to optionally skip role generation

## Changes
* Add `schema_roles` attribute to `SchemaBlueprint`. This should be a list of strings accepting values `["owner", "read", "write"]`
* Update `SchemaParser` to accept `schema_roles` attribute in YAML config
* Update `SchemaRoleResolver` to optionally skip generation of some schema roles based on the value set for `schema_roles` attribute in `SchemaBlueprint`

### CODE REVIEW

**Comment by acaruso7** [comment]
My mistake, I had intended to use my fork as the base branch here, updating . . .

**Comment by acaruso7** [comment]
> My mistake, I had intended to use my fork as the base branch here, updating . . .

Closing - nonetheless, curious to hear your thoughts on this change @littleK0i

**Comment by littleK0i** [comment]
Schema roles and future grants are always created in advance due to lack of transaction support for Snowflake DDL commands. Essentially, it is not possible to avoid race condition if you do not use future grants.

"Owner" schema role is essential. Without it all schema objects will be owned by SnowDDL admin role, and only admins will be able to ALTER or DROP these objects. It is relatively difficult to change ownership later on.

You may not need "read" or "write" roles initially, but what if you to start using it later? Currently all you need to do is to update `business_role.yaml`. Everything works out of the box, very easy to understand and maintain.

But if it becomes possible to skip "read" or "write" roles for some schemas, now we are in trouble. We need to remember that not all schemas are equal. Maintaing complex config will become harder and less obvious.

---

In general, if you have performance issues with DDL commands around roles, you probably have waaaay too many schemas, like 200+, and things are overcomplicated already.

But you may try to increase the number of workers with something like `--max-workers 30`.

**Comment by littleK0i** [comment]
So, TL;DR, I consider adding an option to create more custom schema roles or adjusting specific privileges in the existing schema roles. For exampe, when Snowflake adds more future grants or more object types, I might be unaware of it, especially if it is something in private preview.

It might be possible to optimize grants performance a little but by combining multiple commands together into one larger grant.

But skipping schema roles per schema is currently not on the roadmap. It may work on surface, but it leads to many other problems down the line.

On the other hand, you may totally try to tailor SnowDDL for your specific needs and run a patched version internally. It is just a basic Python script, so can be adjusted easily.
